Source/WebCore/dom/FullscreenManager.cpp

```diff
@@ -412,7 +412,8 @@ void FullscreenManager::exitFullscreen(RefPtr<DeferredPromise>&& promise)
         // If there is a pending fullscreen element but no fullscreen element
         // there is a pending task in requestFullscreenForElement(). Cause it to cancel and fire an error
         // by clearing the pending fullscreen element.
-        if (!m_fullscreenElement && m_pendingFullscreenElement) {
+        RefPtr exitedFullscreenElement = fullscreenElement();
```

Comment:
Why not `auto exitedFullscreenElement = protectedFullscreenElement()` ?

Reply:
We generally like to write `RefPtr` instead of `auto` so you can see the lifetime while still not seeing a distractingly long type name, even though for types that are not pointers we are comfortable with `auto`.

Reply:
Similarly, we use `protectedFoo()` functions to *avoid* local variables, not to initialize local variables. There is no perf benefit for initializing a local variable and it is less concise.